include/boost/spirit/home/qi/detail/expectation_failure.hpp

```diff
@@ -0,0 +1,32 @@
+/*=============================================================================
+Copyright (c) 2001-2011 Joel de Guzman
+
+Distributed under the Boost Software License, Version 1.0. (See accompanying
+file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+=============================================================================*/
+#if !defined(SPIRIT_EXPECTATION_FAILURE_JULY_19_2016)
+#define SPIRIT_EXPECTATION_FAILURE_JULY_19_2016
+
+#if defined(_MSC_VER)
+#pragma once
+#endif
+
+#include <stdexcept>
+
+namespace boost { namespace spirit { namespace qi {
+    template <typename Iterator>
+    struct expectation_failure : std::runtime_error
+    {
+        expectation_failure(Iterator first_, Iterator last_, info const& what)
```

Comment:
After that issues showed up, I ran the build on the whole spirit test-suite. Before I did run on expect directive only in order to detect c++11-isms.

I found that the test suite does not build without errors on either toolset msvc-9.0, msvc-12.0 and msvc-14.0. The expect directive's tests pass on those three. The errors are not related to the expect directive and look quite serious. I seem not to be able to attach the output files here, so I will send them to the mailing list.

At the moment I only have the several MS toolsets installed. I understand that I will have to provide non-MS toolsets as well. I will do.